palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/service/DataService.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2019 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.service.palisade.service;
+
+import uk.gov.gchq.palisade.service.Service;
+import uk.gov.gchq.palisade.service.palisade.exception.NoCapacityException;
+import uk.gov.gchq.palisade.service.palisade.request.ReadRequest;
+import uk.gov.gchq.palisade.service.palisade.request.ReadResponse;
+import uk.gov.gchq.palisade.service.request.Request;
+
+import java.util.concurrent.CompletableFuture;
+
+public interface DataService  extends Service {
```

Comment:
This was brought across from the legacy repo in order to allow the creation of a MockDataService as the Data Service code has not been migrated yet. This was done in order to add the ConnectionDetail part of the resources Map for testing.

This may need to be refactored once the Data Service has been fully migrated to the services repo.